tests/test.cpp

```diff
@@ -2515,6 +2521,45 @@ void ParseUnionTest() {
           true);
 }
 
+void ValidSameNameDifferentNamespaceTest() {
+  // Duplicate table names in different namespaces must be parsable
+  flatbuffers::Parser p;
+  TEST_ASSERT(p.Parse("namespace A; table X {} namespace B; table X {}"));
```

Comment:
optionally: it is possible to parse with a temporary parser: `flatbuffers::Parser().Parse(....)`.